spack-enable.bash

```diff
@@ -18,15 +18,53 @@
 # adding that section, Spack replaces what was in that section with the new
 # value.
 
-# Disable ~/.spack because we have seen compiler misconfiguration by users.
-# This also disables /etc/spack .
-export SPACK_DISABLE_LOCAL_CONFIG="true"
-
-# Setting SPACK_DISABLE_LOCAL_CONFIG disables the following:
-#export SPACK_SYSTEM_CONFIG_PATH=""
-#export SPACK_USER_CONFIG_PATH=""
-
 # TODO: Decide on the version of Python that we approve.
 #export SPACK_PYTHON=
 
-. $(dirname $BASH_SOURCE)/../spack/share/spack/setup-env.sh
+CONFIGDIR="$(realpath $(dirname $BASH_SOURCE))"
+
+# NOTE: A lot of ugly code below to avoid setting variables in the
+#       user's environment. e.g. Avoid defining functions.
+
+if [ "$#" -eq 0 ]
+then
+    # Disable ~/.spack because we have seen compiler misconfiguration by
+    # users. This also disables /etc/spack .
+    export SPACK_DISABLE_LOCAL_CONFIG="true"
+
+    # Setting SPACK_DISABLE_LOCAL_CONFIG disables the following:
+    # export SPACK_SYSTEM_CONFIG_PATH=""
+    # export SPACK_USER_CONFIG_PATH=""
+    echo "Using configuration in spack/etc/spack"
+
+    # NOTE: If using symlinks is undesirable, the following commands can
+    #       be used to construct the name of the relevant config directory.
+    # git -C ${prefix}/spack branch
+    # hostname --fqdn
+
+# NOTE: This option is for CI/CD
+elif [ "$#" -eq 1 ]
+then
+    CONFIG="${CONFIGDIR}/${1}"
+    if [ ! -d "${CONFIG}" ]
+    then
+        echo "Directory ${CONFIG} does not exist."
+        unset CONFIG
+        unset CONFIGDIR
+        return 1
+    fi
+
+    echo "Using configuration in ${CONFIG}"
+    export SPACK_DISABLE_LOCAL_CONFIG="false"
+    export SPACK_USER_CONFIG_PATH="${CONFIG}"
+    export SPACK_SYSTEM_CONFIG_PATH="${CONFIG}"
+    unset CONFIG
+
+else
+    echo "Usage: . ${PROGNAME} [<spack-version>/{ci,gadi}]"
```

Comment:
I don't understand this intended usage. Is it expected to be able to invoke it like so:
```
. spack-enable.bash 0.22/ci
```
for non-CI/CD?

Or
```
. spack-enable.bash 0.22/gadi
```
for CI/CD?

I'm thinking of the intention is for a version to be specified only for CI/CD then
```
. spack-enable.bash 0.22
```
would be sufficient, and you could add `ci` to the `0.22` path automatically. 

Or change the help to
```
echo "Usage: . ${PROGNAME} [<spack-version>/ci]"
```

Am I understanding the intention of this correctly?

Reply:
For CI: `. spack-enable.bash 0.22/ci`
For CD: `. spack-enable.bash 0.22/gadi`